src/mps/cumps.jl

```diff
@@ -9,9 +9,20 @@ end
 cu(ψ::MPS) = cuMPS(ψ)
 
 cuMPS() = MPS() 
-function cuMPS(sites) # random MPS
+function cuMPS(::Type{T}, sites::Vector{<:Index}; linkdims::Integer=1) where {T<:Number}
   N = length(sites)
   v = Vector{ITensor}(undef, N)
+  if N == 1
+    v[1] = emptyITensor(T, sites[1])
```

Comment:
Maybe should be `cuITensor`?

Also, we should discuss adapting the `EmptyStorage` type for use with GPU. In principle it should work fine for GPU. On CPU, the default constructor is like this:
```julia
julia> i = Index(2);

julia> A = ITensor(i);

julia> typeof(storage(A))
EmptyStorage{ITensors.NDTensors.EmptyNumber, Dense{ITensors.NDTensors.EmptyNumber, Vector{ITensors.NDTensors.EmptyNumber}}}
```
We could make it so that there is a corresponding GPU storage type, such as:
```julia
EmptyStorage{ITensors.NDTensors.EmptyNumber, Dense{ITensors.NDTensors.EmptyNumber, CuVector{ITensors.NDTensors.EmptyNumber}}}
```
(change `Vector` to `CuVector`). Then when an element is set, the tensor would allocate on GPU. This would make it so we wouldn't need all of this code branching and could use more generic code for CPU and GPU. For example, if the default MPS on GPU was made with `EmptyStorage` ITensors like it is on CPU, we could do a "cheap"/"lazy" transfer from CPU to GPU and use a simple definition like:
```julia
cuMPS(::Type{T}, sites::Vector{<:Index}; linkdims::Integer=1) where {T<:Number} = cu(MPS(T, sites; linkdims=linkdims))
```

Reply:
Agreed re: `EmptyStorage`. In general I think it might make sense to widen a lot of the types in the CPU code from `Vector` -> `AbstractVector`, which would allow me to delete a lot of code in this package.